cpp/src/aztec3/circuits/rollup/base/native_base_rollup_circuit.cpp

```diff
@@ -159,12 +158,15 @@ std::array<NT::fr, 3> calculate_new_subtrees(BaseRollupInputs baseRollupInputs,
 
         for (size_t j = 0; j < new_commitments.size(); j++) {
             // todo: batch insert
-            commitments_tree.update_element(i * 4 + j, new_commitments[j]);
+            commitments_tree.update_element(i * KERNEL_NEW_COMMITMENTS_LENGTH + j, new_commitments[j]);
```

Comment:
The subtree to be inserted here need to have the `val`, `next_index`, `next_value`

Reply:
will do in another pr